### DISCUSSION TITLE
Custom transaction descriptions

### DISCUSSION THREAD

**Discussion by dfunnell1** [liedekef/events-made-easy#93] [General]
Would it be possible to customise transaction descriptions and/or retrieve transaction IDs?
I am using sumup payment gateway and I am struggling to see how I can match transactions to individual memberships and bookings. It would be good to be able to include something like a name to the description.
I know there is also a sumup payment notification url in the plugin but i can't find any documentation of what this is for as the url is blank.

**Comment by liedekef** [comment]
The payment notification is for sumup (calling back to EME to notify EME of payment success or not).
The payment description can be changed using filters, an example:

```
function my_eme_member_payment_descr($description,$payment,$gateway_name) {
   if ($gateway_name=="sumup") $description .= "Some extra sumup prefix ";
    return $description;
}
add_filter('eme_member_paymentform_description_filter','my_eme_member_payment_descr',10,3);
```
For more info, see the doc: https://www.e-dynamics.be/wordpress/category/documentation/12-hooks-and-filters/

**Reply by dfunnell1** [reply]
Thanks Franky, My apologies, i hadn't seen about hooks and filters. That said having now read the document on your site and on the wordpress site and ultimately adding the code, i am getting some strange behaviour.
I have added the following code to a child theme "functions.php":
```
function my_eme_member_payment_descr($description,$payment,$gateway_name) {
   if ($gateway_name=="sumup"){
		$member = eme_get_member($payment['member_id']);
		$person = eme_get_person($member['person_id']);
   		$description = sprintf("%s %s Membership",$person['firstname'],substr($person['lastname'],0,1));
    }
   return $description;
}
add_filter('eme_member_paymentform_description_filter','my_eme_member_payment_descr',10,3);
```

The filter is clearly being enabled as the descriptions have changed, however they are now only showing as "Membership" without inclusion of the person's name as specified in the code.

I have checked the array fields against the db and they seem to be accurate from what i can tell, but despite this it does not seem to be pulling the person details.
Apologies for all of the questions, I'm quite new to php. Any ideas would be very much appreciated.

**Reply by liedekef** [reply]
Sorry, the demo code in the doc was still using older code there. Change your $member-call to be:
`    $member = eme_get_member_by_paymentid($payment['id']);`
and try again.

**Reply by dfunnell1** [reply]
Brilliant, that's fixed it. Thank you so much for your help.
